tidy3d/plugins/mode/mode_solver.py

```diff
@@ -164,20 +173,36 @@ def rotate_field_coords(e_field, h_field):
             E *= np.exp(-1j * phi)
             H *= np.exp(-1j * phi)
 
-            # # Handle symmetries
-            # if mode.symmetries[0] != 0:
-            #     E_half = E[:, 1:, ...]
-            #     H_half = H[:, 1:, ...]
-            #     E = np.concatenate((+E_half[:, ::-1, ...], E_half), axis=1)
-            #     H = np.concatenate((-H_half[:, ::-1, ...], H_half), axis=1)
-            # if mode.symmetries[1] != 0:
-            #     E_half = E[:, :, 1:, ...]
-            #     H_half = H[:, :, 1:, ...]
-            #     E = np.concatenate((+E_half[:, :, ::-1, ...], E_half), axis=2)
-            #     H = np.concatenate((-H_half[:, :, ::-1, ...], H_half), axis=2)
+            # Expand symmetries
+            e_full = np.zeros([3] + plane_shape + [1], dtype=np.complex128)
+            h_full = np.zeros([3] + plane_shape + [1], dtype=np.complex128)
+            nx_solver, ny_solver = E.shape[1:3]
+            e_full[:, -nx_solver:, -ny_solver:, :] = E
+            h_full[:, -nx_solver:, -ny_solver:, :] = H
+            Nx, Ny = plane_shape
+            if mode_spec.symmetries[0] != 0:
+                sym_val = mode_spec.symmetries[0]
+                # Pad symmetry-tangential E fields
+                e_full[[1, 2], 1:Nx//2, :, :] = sym_val * e_full[[1, 2], :Nx//2:-1, :, :]
```

Comment:
I feel like we should think about generalizing these a bit in a symmetry padding function, it's a bit unwieldy